linera-service/src/storage.rs

```diff
@@ -189,6 +231,92 @@ impl FromStr for StorageConfig {
                 use_localstack,
             });
         }
+        #[cfg(feature = "scylladb")]
+        if let Some(s) = input.strip_prefix(SCYLLA_DB) {
+            let mut restart_database: Option<bool> = None;
+            let mut host: Option<String> = None;
+            let mut port: Option<String> = None;
+            let mut table_name: Option<String> = None;
+            if !s.is_empty() {
+                let mut parts = s.split(':');
+                loop {
+                    let part = parts.next();
+                    match part {
+                        Some(part_ent) => {
```

Comment:
We generally try to avoid truncated words like `ent`.

Why is it not a for-loop? `for part in parts { .. }`

Reply:
Also, this parser is large enough to deserve a function and a unit test.

Reply:
It is definitely tested with several unit tests. But yes, it can be a separate function.

Reply:
I did not do a for-loop because in the case of entries such as `https:://localhost` I needed to progress in the analysis to the next entry.
Maybe I could have done it via a for loop, but that would not necessarily have been a simplification.

Reply:
You could still use `while Some(part_ent) = parts.next() {` instead of `loop`.

Reply:
> I did not do a for-loop because in the case of entries such as `https:://localhost` I needed to progress in the analysis to the next entry. Maybe I could have done it via a for loop, but that would not necessarily have been a simplification.

That makes sense and is a valid use-case for `while let` indeed

Reply:
Ok, thanks, done.